deployment/certs/generate-test-certs-keyvault.sh

```diff
@@ -20,26 +27,37 @@ mkdir $TEMP_DIR
 
 generate_root_ca_cert() {
     local certName=$1
+    local certDir=$2
 
-    ###########################
+    #########################################################
     # Create a custom CA certificate to be used for 
     # signing both cluster and client the certificate
-    ###########################
-
-    # Create custom CA-Certificate in Key-Vault
-    #CSR=$(az keyvault certificate create --vault-name $keyVault --name $certName --policy "@$certName-cert-policy.json" | jq -r '.csr')
-
-    # generate the CA private key
-    openssl genrsa -out $CERTS_DIR/$certName.key 4096
-    
-    # generate the CA certificate
-    openssl req -new -x509 -key $CERTS_DIR/$certName.key -sha256 -subj "/OU=Test CA Corporation/O=Test CA Corporation/L=London/S=Greater London/C=UK" -days 365 -out $CERTS_DIR/$certName.cert
-    
-    # generate the CA certificate .pfx file
-    openssl pkcs12 -export -out "$CERTS_DIR/$certName.pfx" -inkey "$CERTS_DIR/$certName.key" -in "$CERTS_DIR/$certName.cert" -keypbe NONE -certpbe NONE -passout pass:
-    
-    # Import .pfx into Key-Vault to be safly stored and used by the application
-    CERT_IMPORT_RESPONSE=$(az keyvault certificate import --vault-name $keyVault --name $certName --file $CERTS_DIR/$certName.pfx)
+    #########################################################
+
+    # Query an OAuth2 token
+    token=$(curl -X POST https://login.microsoftonline.com/$tenantId/oauth2/token \
+        -d "grant_type=client_credentials&client_id=${clientId}&client_secret=${clientSecret}&resource=https://vault.azure.net" | jq -r '.access_token')
+
+    # Create a new certificate in Key Vault
+    curl -X POST "https://$keyVault.vault.azure.net/certificates/$certName/create?api-version=7.2" \
+        -H "Authorization: Bearer $token" -H "content-type: application/json" \
+        --data "@$certName-cert-policy-curl.json"
```

Comment:
we can use the az rest to skip the access token acquiring step
az rest \
     --method post \
     --body @$certName-cert-policy.json \
     --resource "https://vault.azure.net" \
     --headers '{"content-type":"application/json"}' \
     --uri "https://$keyVault.vault.azure.net/certificates/$certName/create" \
     --uri-parameters 'api-version=7.2'

Reply:
Great feedback, @Aymalla!